https://bugs.gentoo.org/424954

From c5fe9f7b723f949457263ef8e22ab807d5b549ce Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Thu, 5 Jul 2012 23:19:09 -0400
Subject: [PATCH] include sys/resource.h where needed

We use functions from sys/resource.h in misc applets, but don't include
the header.  This breaks building with newer glibc versions, so add the
include where needed.

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
---
 loginutils/passwd.c      |    1 +
 miscutils/time.c         |    1 +
 networking/inetd.c       |    1 +
 networking/ntpd.c        |    1 +
 networking/ntpd_simple.c |    1 +
 runit/chpst.c            |    1 +
 shell/shell_common.c     |    1 +
 7 files changed, 7 insertions(+)

diff --git a/loginutils/passwd.c b/loginutils/passwd.c
index b83db00..a7006f0 100644
--- a/loginutils/passwd.c
+++ b/loginutils/passwd.c
@@ -15,6 +15,7 @@
 
 #include "libbb.h"
 #include <syslog.h>
+#include <sys/resource.h> /* setrlimit */
 
 static void nuke_str(char *str)
 {
diff --git a/miscutils/time.c b/miscutils/time.c
index 945f15f..ffed386 100644
--- a/miscutils/time.c
+++ b/miscutils/time.c
@@ -16,6 +16,7 @@
 //usage:     "\n	-v	Verbose"
 
 #include "libbb.h"
+#include <sys/resource.h> /* getrusage */
 
 /* Information on the resources used by a child process.  */
 typedef struct {
diff --git a/networking/inetd.c b/networking/inetd.c
index 1308d74..00baf69 100644
--- a/networking/inetd.c
+++ b/networking/inetd.c
@@ -165,6 +165,7 @@
 //usage:     "\n		(default: 0 - disabled)"
 
 #include <syslog.h>
+#include <sys/resource.h> /* setrlimit */
 #include <sys/un.h>
 
 #include "libbb.h"
diff --git a/networking/ntpd.c b/networking/ntpd.c
index 72e9d0b..5b92db6 100644
--- a/networking/ntpd.c
+++ b/networking/ntpd.c
@@ -46,6 +46,7 @@
 #include "libbb.h"
 #include <math.h>
 #include <netinet/ip.h> /* For IPTOS_LOWDELAY definition */
+#include <sys/resource.h> /* setpriority */
 #include <sys/timex.h>
 #ifndef IPTOS_LOWDELAY
 # define IPTOS_LOWDELAY 0x10
diff --git a/networking/ntpd_simple.c b/networking/ntpd_simple.c
index 4ad44e4..1b7c66b 100644
--- a/networking/ntpd_simple.c
+++ b/networking/ntpd_simple.c
@@ -7,6 +7,7 @@
  */
 #include "libbb.h"
 #include <netinet/ip.h> /* For IPTOS_LOWDELAY definition */
+#include <sys/resource.h> /* setpriority */
 #ifndef IPTOS_LOWDELAY
 # define IPTOS_LOWDELAY 0x10
 #endif
diff --git a/runit/chpst.c b/runit/chpst.c
index ac296ba..ed72c8b 100644
--- a/runit/chpst.c
+++ b/runit/chpst.c
@@ -91,6 +91,7 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 //usage:     "\n			a SIGXCPU after N seconds"
 
 #include "libbb.h"
+#include <sys/resource.h> /* getrlimit */
 
 /*
 Five applets here: chpst, envdir, envuidgid, setuidgid, softlimit.
diff --git a/shell/shell_common.c b/shell/shell_common.c
index 51c92d6..780e27e 100644
--- a/shell/shell_common.c
+++ b/shell/shell_common.c
@@ -18,6 +18,7 @@
  */
 #include "libbb.h"
 #include "shell_common.h"
+#include <sys/resource.h> /* getrlimit */
 
 const char defifsvar[] ALIGN1 = "IFS= \t\n";
 
-- 
1.7.9.7

